shopfloor/services/checkout.py

```diff
@@ -970,6 +970,16 @@ def _scan_package_find(self, picking, barcode, search_types=None):
             ),
         )
 
+    def _find_line_to_increment(self, product_lines):
+        # Return the first line for the scanned product
+        # which still has some qty todo.
+        # If none are found, return the first line for that product.
+        line = next(
+            (line for line in product_lines if line.qty_done < line.product_uom_qty),
+            None,
+        )
+        return line or fields.first(product_lines)
```

Comment:
You could have put the default value in next
```suggestion
        return next(
            (line for line in product_lines if line.qty_done < line.product_uom_qty),
            fields.first(product_lines),
        )
```